perf-tool/src/agentOptions.cpp

```diff
@@ -103,6 +104,7 @@ void modifyMonitorEvents(const std::string& function, const std::string& command
     {
         if (verbose >= Verbose::INFO)
             printf("Processing MonitorEvents stop command\n");
+        m_monitor.clear();
```

Comment:
We need to protect against races for accessing this hashtable. This means that we have to use the same monitor `monitorMapMutex` used for adding to the hash. This also means that we need to make that monitor a global.

Reply:
@mpirvu - PTAL